resource/cmake/DdsFunctions.cmake

```diff
@@ -13,76 +13,104 @@
 #                                                                             #
 ###############################################################################
 
-function(rtigw_connextdds_run_codegen)
+#[[
+connextdds_rtiddsgen_run
+----------------
+
+ * Brief: generate code using rtiddsgen from an input IDL file
+ * Params:
+ ** IDL_FILE: the IDL filename that will be used to generate code.
+ ** LANG: the language to generate source files for. Only C, C++ and C++11 are
+          currently supported.
+ ** OUTPUT_DIR: the directory where to put generated files.
+ ** FILES: list of files to be copied with relative path form
+           ${CMAKE_CURRENT_SOURCE_DIR}.
+
+ * The variable IDL_NAME shall be defined with the name of the IDL (wihtout
+   extension)
+
+ * How to use it:
+
+    connextdds_rtiddsgen_run(
+        IDL_FILE "${IDL_DIR}/${IDL_NAME}.idl"
+        LANG "C++11"
+        OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/idl"
+        EXTRA_ARGS
+            "-unboundedSupport"
+            "-qualifiedEnumerator"
+    )
+]]
+
+function(connextdds_rtiddsgen_run)
     set(_BOOLEANS)
-    set(_SINGLE_VALUE_ARGS IDL_FILE LANGUAGE OUTPUT_DIR)
-    set(_MULTI_VALUE_ARGS INPUT_PARAMS)
+    set(_SINGLE_VALUE_ARGS IDL_FILE LANG OUTPUT_DIR)
+    set(_MULTI_VALUE_ARGS EXTRA_ARGS)
 
-    cmake_parse_arguments(_CONNEXTDDS
+    cmake_parse_arguments(_args
         "${_BOOLEANS}"
         "${_SINGLE_VALUE_ARGS}"
         "${_MULTI_VALUE_ARGS}"
         ${ARGN}
     )
 
-    if (NOT _CONNEXTDDS_IDL_FILE)
+    if (NOT _args_IDL_FILE)
         message(FATAL_ERROR "No IDL_FILE variable when calling codegen.")
     endif()
-    if (NOT _CONNEXTDDS_LANGUAGE)
-        message(FATAL_ERROR "No LANGUAGE variable when calling codegen.")
+    if (NOT _args_LANG)
+        message(FATAL_ERROR "No LANG variable when calling codegen.")
     endif()
-    if (NOT _CONNEXTDDS_OUTPUT_DIR)
+    if (NOT _args_OUTPUT_DIR)
         message(FATAL_ERROR "No OUTPUT_DIR variable when calling codegen.")
     endif()
 
-    if (NOT ${_CONNEXTDDS_LANGUAGE} MATCHES "^C(\\+\\+|$)(11|$)")
-        message(FATAL_ERROR "Calling rtiddsgen with language <${_CONNEXTDDS_LANGUAGE}>."
+    if (NOT ${_args_LANG} MATCHES "^C(\\+\\+|$)(11|$)")
+        message(FATAL_ERROR "Calling rtiddsgen with language <${_args_LANG}>."
             "This language is unsupported, only C, C++ and C++11 are available")
     endif()
 
     # Source files to be generated C
     set(GENERATED_C_FILES
-        "${_CONNEXTDDS_OUTPUT_DIR}/${IDL_NAME}.c"
-        "${_CONNEXTDDS_OUTPUT_DIR}/${IDL_NAME}Plugin.c"
-        "${_CONNEXTDDS_OUTPUT_DIR}/${IDL_NAME}Support.c"
+        "${_args_OUTPUT_DIR}/${IDL_NAME}.c"
```

Comment:
This function should not depend on an externally defined variable, since it accepts arguments.

The base name of the IDL file should be computed automatically by the function, e.g.:

```cmake
get_filename_component(idl_file "${_args_IDL_FILE}" NAME)
string(REGEX REPLACE "\.idl$" "" idl_name "${idl_file}")
```

Reply:
this should be gone after using `rticonnextdds-cmake-utils`